charts/src/main/java/hu/ma/charts/line/LineChart.kt

```diff
@@ -0,0 +1,266 @@
+import androidx.compose.foundation.gestures.detectTapGestures
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.geometry.Offset
+import androidx.compose.ui.graphics.Brush
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.Path
+import androidx.compose.ui.graphics.nativeCanvas
+import androidx.compose.ui.graphics.toArgb
+import androidx.compose.ui.input.pointer.pointerInput
+import androidx.compose.ui.platform.LocalConfiguration
+import androidx.compose.ui.platform.LocalDensity
+import hu.ma.charts.line.data.DrawAxis
+import hu.ma.charts.line.data.LineChartData
+import kotlin.math.abs
+import kotlin.math.max
+
+@Composable
+fun LineChart(
+  modifier: Modifier = Modifier,
+  data: LineChartData,
+  onDrillDown: ((x: Int, series: List<LineChartData.SeriesData>) -> Unit)? = null
+) {
+
+  val maxXValues = data.series.maxOf { it.points.maxOf { point -> point.x } }
+
+  val maxYValue = max(data.series.maxOf { it.points.maxOf { point -> point.value } },
+    if (data.yLabels.isNotEmpty()) data.yLabels.maxOf { it.atValue } else 0f)
+
+  var drillDownPoint by remember { mutableStateOf<Float?>(null) }
+
+  val screenWidth = with(LocalDensity.current) {
+    (LocalConfiguration.current.screenHeightDp * density)
+  }
+
+  val animatedDrillDownX by animateFloatAsState(targetValue = drillDownPoint ?: screenWidth)
+
+  Canvas(modifier = modifier
+    .background(data.chartColors.background)
+    .pointerInput("drag") {
+      if (onDrillDown != null) {
+        detectHorizontalDragGestures(
+          onHorizontalDrag = { change, _ ->
+            if (change.position.x >= 0 && change.position.x <= this.size.width) {
+              drillDownPoint = change.position.x
+            }
+          },
+          onDragEnd = {
+            val xinterval = this.size.width / maxXValues
+            val snapToPoint = snapToPoints(xinterval, drillDownPoint ?: 0f, data.series)
+            if (snapToPoint != null) {
+              drillDownPoint = ((snapToPoint.x) * xinterval).toFloat()
+              onDrillDown(snapToPoint.x, data.series)
+            }
+          })
+      }
+    }
+    .pointerInput("tap") {
+      if (onDrillDown != null) {
+        detectTapGestures(onTap = {
+          val xinterval = this.size.width / maxXValues
+          val snapToPoint = snapToPoints(xinterval, it.x, data.series)
+          if (snapToPoint != null) {
+            drillDownPoint = ((snapToPoint.x) * xinterval).toFloat()
+            onDrillDown(snapToPoint.x, data.series)
+          }
+        })
+      }
+    }, onDraw = {
+
+    val axisLabelPaint = Paint().apply {
+      textSize = data.axisTextSize.toPx()
+      color = data.chartColors.label.toArgb()
+      typeface = data.axisTypeface
+      isAntiAlias = true
+    }
```

Comment:
Would it be possible to use `Text` instead of painting on Canvas? I think this would be a bit better in terms of customisability with existing Compose primitives

Reply:
Might be - will look into it 👍

Reply:
Have not found a good way to use Text instead of drawing text on Canvas, so will keep it as is for now.